lib/supavisor/monitoring/telem.ex

```diff
@@ -3,69 +3,102 @@ defmodule Supavisor.Monitoring.Telem do
 
   require Logger
 
-  alias Supavisor, as: S
+  @metrics_disabled Application.compile_env(:supavisor, :metrics_disabled, false)
 
-  @spec network_usage(:client | :db, S.sock(), S.id(), map()) :: {:ok | :error, map()}
+  defmacro if_metrics_enabled(do: block) do
+    if @metrics_disabled do
+      quote do
+        nil
+      end
+    else
+      block
+    end
+  end
+
+  defmacro network_usage_disable(do: block) do
+    if @metrics_disabled do
+      quote do
+        {:ok, %{recv_oct: 0, send_oct: 0}}
+      end
+    else
+      block
+    end
+  end
+
+  @spec network_usage(:client | :db, Supavisor.sock(), Supavisor.id(), map()) ::
+          {:ok | :error, map()}
   def network_usage(type, {mod, socket}, id, _stats) do
```

Comment:
I think that I would put whole function inside of `if` and if disabled, then I would create NOP function.